arctic/store/_ndarray_store.py

```diff
@@ -404,8 +405,9 @@ def write(self, arctic_lib, version, symbol, item, previous_version, dtype=None)
         if previous_version:
             if 'sha' in previous_version \
                     and self.checksum(item[:previous_version['up_to']]) == previous_version['sha']:
-                #The first n rows are identical to the previous version, so just append.
-                self._do_append(collection, version, symbol, item[previous_version['up_to']:], previous_version)
+                # The first n rows are identical to the previous version, so just append.
+                # Do a 'dirty' append (i.e. concat & start from a new base version) for safety
+                self._do_append(collection, version, symbol, item[previous_version['up_to']:], previous_version, dirty_append=True)
```

Comment:
Yup, that doesn't seem unreasonable. I think the only difference is that it just concats any previous appends, rather than the whole thing, but I've got no evidence that is worthwhile optimisation.